src/lib/routing-tables.js

```diff
@@ -0,0 +1,219 @@
+    })
+  }
+
+  _removeRoute (ledgerB, ledgerC, connectorFromBToC) {
+    this.eachSource((tableFromA, ledgerA) => {
+      tableFromA.removeRoute(ledgerC, connectorFromBToC)
+    })
+  }
+
+  removeExpiredRoutes () {
+    const now = Date.now()
+    this.eachRoute((routeFromAToB, ledgerA, ledgerB, nextHop) => {
+      if (routeFromAToB.info.expiresAt && routeFromAToB.info.expiresAt < now) {
+        this._removeRoute(ledgerA, ledgerB, nextHop)
+      }
+    })
+  }
+
+  /**
+   * @param {function(tableFromA, ledgerA)} fn
+   */
+  eachSource (fn) { _.forEach(this.sources, fn) }
+
+  /**
+   * @param {function(routeFromAToB, ledgerA, ledgerB, nextHop)} fn
+   */
+  eachRoute (fn) {
+    this.eachSource((tableFromA, ledgerA) => {
+      for (const ledgerB of tableFromA.destinations.keys()) {
+        const routesFromAToB = tableFromA.destinations.get(ledgerB)
+        for (const nextHop of routesFromAToB.keys()) {
+          const routeFromAToB = routesFromAToB.get(nextHop)
+          fn(routeFromAToB, ledgerA, ledgerB, nextHop)
+        }
+      }
+    })
+  }
+
+  /**
+   * @returns {Routes}
+   */
+  toJSON () {
+    const routes = []
+    this.eachSource((table, sourceLedger) => {
+      for (const destinationLedger of table.destinations.keys()) {
+        const routesByConnector = table.destinations.get(destinationLedger)
+        const combinedRoute = combineRoutesByConnector(routesByConnector)
+        routes.push({
+          source_ledger: sourceLedger,
+          destination_ledger: destinationLedger,
+          connector: this.baseURI,
+          points: combinedRoute.getPoints(),
+          min_message_window: combinedRoute.info.minMessageWindow,
+          source_account: this._getAccount(this.baseURI, sourceLedger)
+        })
+      }
+    })
+    return routes
+  }
+
+  _getAccount (connector, ledger) {
+    return this.accounts[connector + ';' + ledger]
+  }
+
+  _getLocalRoute (ledgerA, ledgerB) {
+    const routesFromAToB = this.sources[ledgerA].destinations.get(ledgerB)
+    if (!routesFromAToB) return
+    return routesFromAToB.get(this.baseURI)
+  }
+
+  /**
+   * Find the best intermediate ledger (`ledgerB`) to use after `ledgerA` on
+   * the way to `ledgerC`.
+   * This connector must have `[ledgerA, ledgerB]` as a pair.
+   *
+   * @param {URI} ledgerA
+   * @param {URI} ledgerC
+   * @param {String} finalAmount
+   * @returns {Object}
+   */
+  findBestHopForDestinationAmount (ledgerA, ledgerC, finalAmount) {
+    const nextHop = this._findBestHopForDestinationAmount(ledgerA, ledgerC, +finalAmount)
+    if (!nextHop) return
+    const ledgerB = nextHop.info.nextLedger
+    const routeFromAToB = this._getLocalRoute(ledgerA, ledgerB)
+    const isLocal = this.baseURI === nextHop.bestHop
+    return {
+      connector: nextHop.bestHop,
+      sourceLedger: ledgerA,
+      // Prevent 'BigNumber Error: new BigNumber() number type has more than 15 significant digits'
+      sourceAmount: nextHop.bestCost.toFixed(15),
```

Comment:
If you call the BigNumber constructor with a string you're not limited in the amount of precision (from [docs](https://mikemcl.github.io/bignumber.js/#bignumber): "Values of type number with more than 15 significant digits are considered invalid (if ERRORS is true) as calling toString or valueOf on such numbers may not result in the intended value. **There is no limit to the number of digits of a value of type string** (other than that of JavaScript's maximum array size).")